lib/writeBinaryGltf.js

```diff
@@ -10,18 +10,25 @@ function writeBinaryGltf(gltf, outputPath, createDirectory, callback) {
     // Correct output path extension if necessary
     var outputExtension = path.extname(outputPath);
     if (outputExtension !== '.glb') {
-        outputPath = path.basename(outputPath, outputExtension) + '.glb';
+        outputPath = path.join(path.dirname(outputPath), path.basename(outputPath) + '.glb');
     }
     // Create the output directory if specified
     if (createDirectory) {
         outputPath = path.join(path.dirname(outputPath), 'output', path.basename(outputPath));
-        mkdirp.sync(path.dirname(outputPath));
     }
+    mkdirp.sync(path.dirname(outputPath));

```

Comment:
This change includes all the deleted specs. I thought it was weird that the pipeline won't export a gltf file unless `createDirectory` is true, especially since there are cases where you don't want the file to be put in an output folder.